### DISCUSSION TITLE
getting started with environment (would like to look into #162)

### DISCUSSION THREAD

**Discussion by traycerb** [neutrinolabs/xrdp#2513] [General]
the following issue with mouse double press [xorgxrdp's xrdpmouse ](https://github.com/neutrinolabs/xorgxrdp/issues/162) is frustrating enough that I'd like to figure out what's causing it and fix it possible.

I haven't done any serious programming in more than a decade, but I know vim/make/gcc.  I cloned and compiled the xrdp and xorgxrdp repositories on WSL2, but in terms of how to move forward with debugging this, not sure where to start in terms of setting up the environment.  can anyone give any basic pointers or examples of your workflow and how you are debugging?

**Comment by matt335672** [comment]
@traycerb - nice discussion topic.

I don#t work a lot with `xorgxrdp` myself as I'm concentrating more on the issues around desktop integration.

For xrdp on Ubuntu I build with these CFLAGS:-

```
export CFLAGS="-g -fvar-tracking -Wl,-z,now"
```

and with the `--enable-devel-all` configure option.

For debugging, I generally use `gdb -p <pid>` to attach to the process in question, set breakpoints, etc.

You should be able to debug `xorgxrdp.so` by attaching a debugger to the `Xorg` process and working in the same way. The xorgxrdp modules will need to be build with debugging symbols, and preferably with optimisation disabled.
